### ISSUE TITLE
30fps performance on RTX3070

### ISSUE THREAD

**Issue by batselem** [haotian-liu/yolact_edge#92]
Hi

You claimed that yolact edge runs 67fps on RTX 2080 Ti without TensorRT on your paper. But, I tested it on RTX 3070 graphic card and obtained 30fps. I checked that [RTX3070 and RTX2080 Ti](https://www.gpucheck.com/compare/nvidia-geforce-rtx-2080-ti-vs-nvidia-geforce-rtx-3070/intel-core-i7-8700k-3-70ghz-vs-intel-core-i7-8700k-3-70ghz/) are quite similar in performance. Also, I am using Windows machine but I believe it is not the reason why the performance is 2x times lower than what you claimed. I appreciate your feedback.

Thanks.

**Comment by haotian-liu** [comment]
Hi, please provide the command you are using.

**Comment by batselem** [comment]
Thank you for your reply.  I am using the following script you provided in Google Colab. I have added a few lines to read video and measure the time with a loop.

```
from eval import *
from utils.logging_helper import setup_logger
import numpy as np
from eval import args
import cv2
import time

parse_args(["--config=yolact_edge_config", "--calib_images=../calib_images"])

setup_logger(logging_level=logging.INFO)
logger = logging.getLogger("yolact.eval")


args.trained_model = "./weights/yolact_edge_54_800000.pth"
args.yolact_transfer = True
torch.set_default_tensor_type('torch.cuda.FloatTensor')

logger.info('Loading model...')
net = Yolact(training=False)
net.load_weights(args.trained_model, args=args)
net.eval()
logger.info('Model loaded.')

net.detect.use_fast_nms = args.fast_nms
cfg.mask_proto_debug = args.mask_proto_debug

args.score_threshold = 0.15
args.top_k = 15


cap = cv2.VideoCapture('E:/Projects/yolact_edge/traffic_y_Trim.mp4')
extras = {"backbone": "full", "interrupt": False, "keep_statistics": False, "moving_statistics": None}

ret, img = cap.read() # first frame to get size
height, width, channels = img.shape
fourcc = cv2.VideoWriter_fourcc(*'MP4V')

out = cv2.VideoWriter('output.mp4', fourcc, 20.0, (width, height), True)

frame_count = 0
total_runtime = 0.
while cap.isOpened():
    ret, img = cap.read()
    # if frame is read correctly ret is True
    if not ret:
        print("Can't receive frame (stream end?). Exiting ...")
        break

    frame = torch.from_numpy(img).cuda().float()
    batch = FastBaseTransform()(frame.unsqueeze(0))

    # Inference
    start_time = time.time()
    with torch.no_grad():
    	preds = net(batch, extras=extras)["pred_outs"]
    
    end_time = time.time()

    total_runtime += (end_time - start_time)

    img_numpy = prep_display(preds, frame, None, None, class_color=True, undo_transform=False, mask_alpha=0.6)

    frame_count += 1
    fps = float(frame_count) / total_runtime

    cv2.putText(img_numpy, 'Count:' + str(frame_count),(10, height - 20), cv2.FONT_HERSHEY_SIMPLEX, 1,(255,255,255),2)
    cv2.putText(img_numpy, str(int(fps)) + ' fps', (10, 30), cv2.FONT_HERSHEY_SIMPLEX, 1,(255,255,255),2)

    cv2.imshow('result', img_numpy)
    out.write(img_numpy)

    if frame_count > 840:
    	break

    if cv2.waitKey(1) == ord('q'):
        break

cap.release()
out.release()
cv2.destroyAllWindows()


```

**Comment by haotian-liu** [comment]
Because CUDA is asynchronous, thus this measurement might not be accurate. Please use `--benchmark` option instead on the validation set.

**Comment by batselem** [comment]
I will check it with `--benchmark` option. Please don't close the issue. I want to report the result and discuss it with you. Thanks.

**Comment by batselem** [comment]
I tested it using benchmark without tensorrt but I got 29.59 fps on RTX3070. Will you check the following result?

```
python eval.py --trained_model=weights/yolact_edge_54_800000.pth --benchmark --max_images=1000 --disable_tensorrt
Config not specified. Parsed yolact_edge_config from the file name.

./data/coco/images/
loading annotations into memory...
Done (t=0.60s)
creating index...
index created!
[04/07 09:56:42 yolact.eval]: Loading model...
[04/07 09:56:44 yolact.eval]: Model loaded.

Processing Images  ██████████████████████████████   1000 /   1000 (100.00%)    29.59 fps

Stats for the last frame:

      Name      | Time (ms)
----------------+------------
  Network Extra |     0.1605
       backbone |    11.7493
            fpn |     0.6823
          proto |     0.5993
     pred_heads |     2.1502
     makepriors |     0.0071
         Detect |    12.1299
    Postprocess |     0.4812
           Sync |     0.0259
----------------+------------
          Total |    27.9857

Average: 29.59 fps, 33.79 ms
```

**Comment by haotian-liu** [comment]
Sorry, since I do not own a RTX 3070, and I did not have Windows-based Python environment, I could not provide much of the information on what we should expect on your case. It could be because of Windows as discussed https://github.com/dbolya/yolact/issues/10, and CPU will also affect the inference speed. I would recommend you to perform experiments on Ubuntu to see if it is OS-related issue.

**Comment by batselem** [comment]
But it is 2 times lower than your results. I understand there is OS difference but still shouldn't affect the result that much. Is there any speed difference between yolact_edge_54_800000.pth and yolact_edge_vid_resnet50_847_50000.pth models?

**Comment by haotian-liu** [comment]
Oh I see you are evaluating on MS-COCO instead of YouTube-VIS. For YouTube-VIS (video dataset), we have video-based optimization, while for COCO datasets (image dataset), we do not. Please refer to Table VI for the FPS of COCO on RTX 2080 Ti.

**Comment by batselem** [comment]
I see. I will try YouTube-Vis dataset. Thank you for your quick responses. Really appreciated.

**Comment by batselem** [comment]
I am setting an ubuntu machine to test your model. I will let you know my results.

**Comment by haotian-liu** [comment]
I am closing this issue for now due to inactivity. If there are any further questions that I could help with, please feel free to reopen, thanks.
